R/serializer.R

```diff
@@ -17,30 +8,267 @@ NULL
 #'
 #' @param name The name of the serializer (character string)
 #' @param serializer The serializer to be added.
+#' @param verbose Logical value which determines if a message should be printed when overwriting serializers
 #'
 #' @export
-addSerializer <- function(name, serializer){
-  if (is.null(.globals$serializers)){
-    .globals$serializers <- list()
-  }
-  if (!is.null(.globals$serializers[[name]])){
-    stop ("Already have a serializer by the name of ", name)
+addSerializer <- function(name, serializer, verbose = TRUE) {
+  if (!is.null(.globals$serializers[[name]])) {
+    if (isTRUE(verbose)) {
+      warning("Overwriting serializer: ", name)
```

Comment:
Maybe this should be a `message()` (since there are scenarios that are no real user 'mistake')